src/icarus/datasetgenerator.py

```diff
@@ -0,0 +1,210 @@
+import random
+import math 
+from pathlib import Path
+from icarus import MooseSetup
+from mooseherder import (MooseHerd,
+                         MooseRunner,
+                         InputModifier,
+                         DirectoryManager,
+                         SweepReader) 
+
+class DatasetGenerator:
+    """Used to generate the unlabelled ground truth, perturbed, and 
+        validation datasets by running the required functions with the necessary parameter
+        names and values, and save paths.
+
+        NB: To run the input file, mooseherder requires that the moose_vars be structured as 
+        [[{param_name: [param_values]}], [{param_name: param_values}], ...] for each parameter.
+    """
+    def __init__(self, moose_runner: MooseRunner, moose_modifier: InputModifier, 
+                 parameters: dict[str, list], output_file_path: str, num_para_runs: int=2) -> None:
+        """__init__
+
+        Parameters
+        ----------
+        moose_runner : MooseRunner
+            Constructed MOOSE runner used to run the input file with modified variables 
+        moose_modifier : InputModifier
+            Used to extract and modify the variables in the input file.
+            Specifies the comment character. Variable definition blocks should begin 
+            #comment character#* and end #comment character#**, e.g. #_* and #** for
+            moose.
+        parameters : dict[str, list]
+            Dictionary containing the name of the parameter and the list of values for
+            that parameter to take for each simulation to be run.
+        output_file_path : str
+            Contains the path to the folder where the datasets and model(s) will be stored.
+        num_para_runs : int, optional
+            Number of parallel runs for running the simulations, by default 2.
+
+        Raises 
+        ----------
+        ValueError
+            If there are no parameters entered.
+        FileNotFoundError
+            If any of the required output file paths don't exist.
+        """
+        self.moose_runner = moose_runner
+        self.moose_modifier = moose_modifier
+        self.num_para_runs = num_para_runs
+
+        if parameters == None or len(parameters) == 0:            
```

Comment:
I *think* this should achieve the same result, as `not []` and `not None` both evaluate to `True`
```suggestion
        if not parameters:            
```